src/chai/ManagedArray.hpp

```diff
@@ -315,7 +320,7 @@ class ManagedArray : public CHAICopyable
    */
   template <bool B = std::is_base_of<CHAICopyable, T>::value,
```

Comment:
I think that would make things more complicated since it would involve more template meta programming than the current case. Furthermore in GEOSX we have a few classes that wrap the ManagedArray and those wrapper classes just inherit from `CHAICopyable` and we don't have to worry about how many nested layers we have.

Reply:
I had proposed a multidimensional array in CHAI in #23 , but the proposal was declined.

Reply:
Eventually it might be nice because then we could move toward picking the striding of the arrays. But that can be a discussion for another time. Great work!